src/FileMutex.php

```diff
@@ -20,39 +19,16 @@ public function __construct(private readonly string $fileName)
 
     public function acquire(): Lock
     {
-        // Try to create the lock file. If the file already exists, someone else
-        // has the lock, so set an asynchronous timer and try again.
+        $f = \fopen($this->fileName, 'c');
         while (true) {
-            try {
-                $file = openFile($this->fileName, 'x');
-
+            if (\flock($f, LOCK_EX|LOCK_NB)) {
                 // Return a lock object that can be used to release the lock on the mutex.
-                $lock = new Lock($this->release(...));
-
-                $file->close();
+                $lock = new Lock(fn () => \flock($f, LOCK_UN));
```

Comment:
This will block for a short duration, no?

Reply:
I considered adding a variant that used parallel but decided that the overhead to communicate to/from the IPC worker will be way worse than that of a single syscall that returns immediately.